azure/compute/vm.go

```diff
@@ -0,0 +1,153 @@
+package compute
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/azure"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+
+	"github.com/pulumi/pulumi-azure-native-sdk/compute"
+	"github.com/pulumi/pulumi-azure-native-sdk/network"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	imageURNSeparator = ":"
+	adminUsername     = "azureuser"
+)
+
+func NewLinuxInstance(e azure.Environment, name, imageUrn, instanceType string, userData pulumi.StringPtrInput) (*compute.VirtualMachine, *network.PublicIPAddress, *network.NetworkInterface, error) {
+	sshPublicKey, err := utils.GetSSHPublicKey(e.DefaultPublicKeyPath())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	linuxOsProfile := compute.OSProfileArgs{
+		ComputerName:  pulumi.String(name),
+		AdminUsername: pulumi.String(adminUsername),
+		LinuxConfiguration: compute.LinuxConfigurationArgs{
+			DisablePasswordAuthentication: pulumi.BoolPtr(true),
+			Ssh: compute.SshConfigurationArgs{
+				PublicKeys: compute.SshPublicKeyTypeArray{
+					compute.SshPublicKeyTypeArgs{
+						KeyData: sshPublicKey,
+						Path:    pulumi.StringPtr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", adminUsername)),
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Super nice !